ahref-backend/common/models/flight.js

```diff
@@ -39,21 +63,22 @@ function doDelete(Flight, ctx, model, next, errorCallback) {
     isolationLevel: sqlFlightReservation.Transaction.READ_COMMITTED,
   }, function(err, tx) {
     if (err) errorCallback(err);
-    // lock car for update
+    // lock flight for update
     postgres.connector.execute(
       'SELECT * FROM sFlight WHERE mongoId = $1 FOR UPDATE;'
       , [ctx.req.params.id], function(err, data) {
         sFlight.findOne({where: {mongoId: ctx.req.params.id}}).then((flight)=>{
 			
```

Comment:
Pass transaction as a parameter to execute function as this doesn't lock the sFlight.
Try to do more optimistic locking, where you lock collection of seats rather than whole sFlight